pyodide-build/pyodide_build/buildpkg.py

```diff
@@ -251,7 +241,9 @@ def download_and_extract(
     src_metadata
         The source section from meta.yaml.
     """
-    response = request.urlopen(src_metadata["url"])
+    # We only call this function when the URL is defined
+    url = cast(str, src_metadata.url)
```

Comment:
Mypy is not able to determine that this is only called with a non None URL and needs some help (this only affects types).

Reply:
I think adding something like `if not src_metadata.url: ...` before this line makes mypy happy.

Reply:
Indeed, but then you need to raise an exception when it is None. Technically that exception is not tested and decreases code coverage (unless one writes a test for it)

This has the same end result but is shorter and without untested lines of code.